protocol_data/rhinofi.py

```diff
@@ -0,0 +1,687 @@
+                "name": "amountNative",
+                "type": "uint256"
+            },
+            {
+                "components": [
+                    {
+                        "internalType": "address",
+                        "name": "target",
+                        "type": "address"
+                    },
+                    {
+                        "internalType": "uint256",
+                        "name": "value",
+                        "type": "uint256"
+                    },
+                    {
+                        "internalType": "bytes",
+                        "name": "data",
+                        "type": "bytes"
+                    }
+                ],
+                "internalType": "struct BridgeVM.Call[]",
+                "name": "datas",
+                "type": "tuple[]"
+            },
+            {
+                "internalType": "bytes",
+                "name": "ref",
+                "type": "bytes"
+            }
+        ],
+        "name": "withdrawWithData",
+        "outputs": [],
+        "stateMutability": "nonpayable",
+        "type": "function"
+    },
+    {
+        "stateMutability": "payable",
+        "type": "receive"
+    }
+]
+""")
+
+    # deposit and fill contracts may be the same for some protocols
+    if type == 'deposit':
+        return { 0 : contract_abi }
+    elif type == 'fill':
+        return { 0 : contract_abi }
+    
+    return None
+
+def get_supported_chains():
+    # Only chain_ids listed here will be used when scraping data
+    # TODO: add more chains
+    return ['42161', '56', '137', '324', '1101', '10', '59144', '8453', '169', '534352']
+
+def get_deposit_function_filter():
+    # To record deposit transactions specify the function name
+    # NOTE: this is optional and can be left as None
+    return None
+
+def get_deposit_event_filter():
+    # To record deposit events specify the deposit function name
+    # NOTE: either this or the deposit function filter must be set
+    return { 0 : ['BridgedDeposit'] }
+
+def get_fill_function_filter():
+    # To record fill transcations specify the function name
+    # NOTE: to accurately record multiple (attempted or otherwise) fills 
+    # for an order where subsequent fills result in rejected transactions
+    # it is VERY IMPORTANT to add a fill function filter. Only adding a 
+    # fill event filter will not result in reverted txs being picked up
+    # TODO: verify rhino.fi does not have any not-successful fills, because
+    # there is only one filler and it gets the funds from pre-filled pools, related discord chat:
+    # https://discord.com/channels/745570257808130058/745571091438764062/1265924362276438108
+    return { 0 : [] }
+
+def get_fill_event_filter():
+    # To record fill events specify the event name
+    # NOTE: wherever possible please also include the fill function filter above
+    return { 0 : ['BridgedWithdrawal', 'BridgedWithdrawalWithData', 'BridgedWithdrawalWithNative'] }
```

Comment:
Not seeing BridgedWithdrawalWithData in the official repo, did you see this onchain?

Reply:
I took it from the ABI (I took the ABI from contract on binance scanner). Maybe I used incorrect ABI? Let me know if there is a better way to find the ABI. Here is how I found it:
![Screenshot 2024-08-01 at 14 59 09](https://github.com/user-attachments/assets/1386fb7d-ea90-4c40-8c51-6edfd174ce3e)
![Screenshot 2024-08-01 at 14 59 32](https://github.com/user-attachments/assets/063af32d-bd14-4485-9c2c-763391c32dd8)
![Screenshot 2024-08-01 at 15 00 10](https://github.com/user-attachments/assets/9a795afc-1e21-4225-b989-2b5ea3b5d60e)

Reply:
Looking at https://bscscan.com/address/0x3337b457dad8339a90a7ba4fdd1d859915b96f7f#code I don't see this event in the contract?

Reply:
Although this makes an interesting point.. perhaps this is an earlier version of the contract which is not longer in use, but we will need the events to backfill the data

Reply:
Found it at https://bscscan.com/address/0x937c802934c71b716879c52c91c2884a36b448fb

Reply:
ok, so we are keeping them here, right?